src/backend/distributed/commands/role.c

```diff
@@ -0,0 +1,258 @@
+ * role statement.
+ */
+List *
+ProcessAlterRoleStmt(AlterRoleStmt *stmt, const char *queryString)
+{
+	ListCell *optionCell = NULL;
+	List *commands = NIL;
+
+	if (!EnableAlterRolePropagation || !IsCoordinator())
+	{
+		return NIL;
+	}
+
+	foreach(optionCell, stmt->options)
+	{
+		DefElem *option = (DefElem *) lfirst(optionCell);
+
+		if (strcasecmp(option->defname, "password") == 0)
+		{
+			Oid roleOid = get_rolespec_oid(stmt->role, true);
+			const char *encryptedPassword = ExtractEncryptedPassword(roleOid);
+
+			if (encryptedPassword != NULL)
+			{
+				Value *encriptedPasswordValue = makeString((char *) encryptedPassword);
+				option->arg = (Node *) encriptedPasswordValue;
+			}
+			else
+			{
+				option->arg = NULL;
+			}
+
+			break;
+		}
+	}
+	commands = list_make1((void *) CreateAlterRoleIfExistsCommand(stmt));
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * CreateAlterRoleIfExistsCommand creates ALTER ROLE command, from the alter role node
+ *  using the alter_role_if_exists() UDF.
+ */
+static const char *
+CreateAlterRoleIfExistsCommand(AlterRoleStmt *stmt)
+{
+	StringInfoData alterRoleQueryBuffer = { 0 };
+	const char *roleName = RoleSpecString(stmt->role);
+	const char *alterRoleQuery = DeparseTreeNode((Node *) stmt);
+
+	initStringInfo(&alterRoleQueryBuffer);
+	appendStringInfo(&alterRoleQueryBuffer,
+					 "SELECT alter_role_if_exists(%s, %s)",
+					 quote_literal_cstr(roleName),
+					 quote_literal_cstr(alterRoleQuery));
+
+	return alterRoleQueryBuffer.data;
+}
+
+
+/*
+ * ExtractEncryptedPassword extracts the encrypted password of a role. The function
+ * gets the password from the pg_authid table.
+ */
+static const char *
+ExtractEncryptedPassword(Oid roleOid)
+{
+	Relation pgAuthId = heap_open(AuthIdRelationId, AccessShareLock);
+	TupleDesc pgAuthIdDescription = RelationGetDescr(pgAuthId);
+	HeapTuple tuple = SearchSysCache1(AUTHOID, roleOid);
+	bool isNull = true;
+	Datum passwordDatum = heap_getattr(tuple, Anum_pg_authid_rolpassword,
+									   pgAuthIdDescription, &isNull);
+
+	heap_close(pgAuthId, AccessShareLock);
+	ReleaseSysCache(tuple);
+
+	return pstrdup(TextDatumGetCString(passwordDatum));
+}
+
+
+/*
+ * GenerateAlterRoleIfExistsCommand generate ALTER ROLE command that copies a role from
+ * the pg_authid table.
+ */
+static const char *
+GenerateAlterRoleIfExistsCommand(HeapTuple tuple, TupleDesc pgAuthIdDescription)
+{
+	char *rolPassword = "";
+	char *rolValidUntil = "infinity";
+	Datum rolValidUntilDatum;
+	Datum rolPasswordDatum;
+	bool isNull = true;
+	Form_pg_authid role = ((Form_pg_authid) GETSTRUCT(tuple));
+	AlterRoleStmt *stmt = makeNode(AlterRoleStmt);
+	const char *rolename = NameStr(role->rolname);
+
+	stmt->role = makeNode(RoleSpec);
+	stmt->role->roletype = ROLESPEC_CSTRING;
+	stmt->role->location = -1;
+	stmt->role->rolename = pstrdup(rolename);
+	stmt->action = 1;
+	stmt->options = NIL;
+
+	stmt->options =
```

Comment:
I know this is annoying, but  the indentation is still not very convenient here :/ Can you check this: https://github.com/citusdata/citus/compare/alter_role_propagation...minor_improvement?expand=1